@here/olp-sdk-dataservice-write/lib/client/VersionedLayerClient.ts

```diff
@@ -47,6 +52,77 @@ export class VersionedLayerClient {
      */
     constructor(private readonly params: VersionedLayerClientParams) {}
 
+    /**
+     * Checks that the datahandle is not used.
+     * Data handles must be unique within the layer across all versions.
+     * In case data handle exists, a new one needs to be generated and checked again until
+     * one is found that is not present in the blob store.
+     * @param request CheckDataExistsRequest with required params.
+     * @param abortSignal An optional signal object that allows you to communicate with a request (such as the `fetch` request)
+     * and, if required, abort it using the `AbortController` object.
+     *
+     * For more information, see the [`AbortController` documentation](https://developer.mozilla.org/en-US/docs/Web/API/AbortController).
+     *
+     * @returns A promise with true if data handle is exists or false if not exists.
+     * Rejects with http response if any http error.
+     */
+    public async checkDataExists(
+        request: CheckDataExistsRequest,
+        abortSignal?: AbortSignal
+    ): Promise<boolean> {
```

Comment:
Boolean does not make much sense here. Either return void means nothing or the HTTP status code which can be only 200 here. For any other HTTP status code, the promise will be rejected.

Reply:
updated, thank you